src/components/toast/toast.js

```diff
@@ -306,17 +328,22 @@ function MdToastProvider($$interimElementProvider) {
     })
     .addPreset('simple', {
       argOption: 'textContent',
-      methods: ['textContent', 'content', 'action', 'highlightAction', 'highlightClass', 'theme', 'parent' ],
+      methods: ['textContent', 'content', 'action', 'actionKey', 'actionHint', 'highlightAction',
+                'highlightClass', 'theme', 'parent', 'dismissHint' ],
       options: /* @ngInject */ function($mdToast, $mdTheming) {
         return {
           template:
             '<md-toast md-theme="{{ toast.theme }}" ng-class="{\'md-capsule\': toast.capsule}">' +
-            '  <div class="md-toast-content">' +
-            '    <span class="md-toast-text" role="alert" aria-relevant="all" aria-atomic="true">' +
+            '  <div class="md-toast-content" role="alert" aria-relevant="all">' +
```

Comment:
Using `role="alert"` isn't really the best pattern for snackbars. `alert` is a more specific type of dialog, which are inherently interuptive, while snackbars are meant to be non-interuptive. 

The recommend approach is to use an `aria-live` region (polite) for the text of the notification (which is what the MatSnackbar does)

Reply:
I looked into using `$mdLiveAnnouncer` which does enable the use of `role="status"` live regions with `aria-live="polite"` but the current implementation of it does not work well with interim elements like toasts.

I'm not 100% sure, but I believe for the current implementation to work properly, the `$mdLiveAnnouncer` needs to be injected in a component that is longer lived. This might work fine if other components are using `$mdLiveAnnouncer` and you add a toast, but it doesn't seem to work if you only add the toast w/o any other components to trigger instantiation earlier.

The [description of the `alert` role](https://www.w3.org/TR/wai-aria-1.1/#alert) does seem to align fairly well with the behavior of snackbars and toasts. 

The [`alertdialog` role](https://www.w3.org/TR/wai-aria-1.1/#alertdialog) does not, as it better fits a modal dialog.

Reply:
Ah, you're right, I mixed up `alert` and `alertdialog`. In that case you should be able to swap `role="alert"` with `aria-live="polite"` and get the same treatment. Not going through the live announcer isn't ideal, but AFAIK it should work most of the time.

Reply:
If I swap to [`role="status"`](https://www.w3.org/TR/wai-aria-1.1/#status) then it should default to `aria-live="polite"`. However, it seems like a snackbar with an interactive button / hotkey would better fit the `alert` role versus the `status` role. 

The other downside here is that `role="status"` is broken in Chromevox until Chrome M70.

It looks like `role="alert" aria-live="polite"` together may be the best fit. I'll do some more testing.

Reply:
OK, adding both isn't a good idea as it causes double reading of the messages. 

However, `polite` in most cases (with our current defaults and demos) causes the toast to be read after the toast times out and disappears. This means that by the time the user heard the instructions for interacting, the toast would be gone.

To mitigate this, I've added instructions to the `hideDelay` docs to recommend setting it to `0` when the toast includes an action. I would like to do this automatically for simple toasts, but I will avoid it for now since it might cause some g3 tests to fail. I've also updated the demos to use `hideDelay(0)` when they have an action.

Reply:
Yeah, I don't believe `role="status"` is widely supported. 

The Material Design team also says that snackbars shouldn't auto-dismiss when they have an action, so that would indeed be the right behavior.